src/domain/core/src/entities/metadata_chain.rs

```diff
@@ -146,6 +152,58 @@ pub trait MetadataChainExt: MetadataChain {
     fn iter_blocks_ref<'a>(&'a self, head: &'a BlockRef) -> DynMetadataStream<'a> {
         self.iter_blocks_interval_ref(head, None)
     }
+
+    async fn accept_interval<'a, E>(
+        &'a self,
+        visitors: VisitorsMutRef<'a, 'a, E>,
+        head: &'a Multihash,
+        tail: Option<&'a Multihash>,
+        ignore_missing_tail: bool,
+    ) -> Result<(), E>
+    where
+        E: Error + From<IterBlocksError>,
+    {
+        let mut decisions = vec![Decision::Next; visitors.len()];
+        let visitor_facade = MetadataChainVisitorFacade::new(&mut decisions, visitors);
+        let block_stream = self.iter_blocks_interval(head, tail, ignore_missing_tail);
+
+        accept_metadata_stream(visitor_facade, block_stream).await
```

Comment:
>  I think accept should be implemented as a simple loop calling get_block.

Done

> I also don't see the point of MetadataChainVisitorFacade as a type either - seems like it could be just inlined here.

Inlined -- Done